### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by Axoden**
refactoring

Petr Janoušek

### CODE REVIEW

File: NNPTPZ1/ComplexNumber.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                RealNumber = conjugate.RealNumber / denominator,
+                ImaginaryNumber = conjugate.ImaginaryNumber / denominator
+            };
+        }
+        public double GetAbsoluteValue()
+        {
+            return Math.Sqrt(RealNumber * RealNumber + ImaginaryNumber * ImaginaryNumber);
+        }
+
+        public double GetAngleInDegrees()
```
Comment (rodi0878) [inline NNPTPZ1/ComplexNumber.cs]: method name doesnt reflect code

File: NNPTPZ1/Polynomial.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections.Generic;
+
+namespace NNPTPZ1
+{
+    internal class Polynomial
+    {
+        /// <summary>
+        /// Coe
```
Comment (rodi0878) [inline NNPTPZ1/Polynomial.cs]: what?

File: NNPTPZ1/Program.cs [hunk 3]
```diff
@@ -23,311 +10,159 @@ namespace NNPTPZ1
...
+        private static int ProcessPixelOfImage(double xmin, double ymin, double xstep, double ystep, Bitmap bmp, Polynomial p, Polynomial pd, List<ComplexNumber> roots, ref int maxid, int i, int j)
         {
-            /// <summary>
-            /// Coe
-            /// </summary>
-            public List<Cplx> Coe { get; set; }
+            ComplexNumber complexNumber = CreateCoordinate(xmin, ymin, xstep, ystep, i, j);
+            float it = FindSolutionNewtonIteration(p, pd, ref complexNumber);
+            int id = FindSolutionRootNumber(roots, ref maxid, complexNumber);
+            Color vv = CreatePixelColor(it, id);
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: vv?

File: NNPTPZ1/Program.cs [hunk 4]
```diff
@@ -23,311 +10,159 @@ namespace NNPTPZ1
...
 
-            /// <summary>
-            /// Evaluates polynomial at given point
-            /// </summary>
-            /// <param name="x">point of evaluation</param>
-            /// <returns>y</returns>
-            public Cplx Eval(double x)
+        private static float FindSolutionNewtonIteration(Polynomial p, Polynomial pd, ref ComplexNumber coordinates)
+        {
+            float it = 0;
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: why float?

File: NNPTPZ1/Program.cs [hunk 5]
```diff
@@ -23,311 +10,159 @@ namespace NNPTPZ1
...
+        }
 
-            public override string ToString()
-            {
-                return $"({Re} + {Imaginari}i)";
-            }
+        private static Color CreatePixelColor(float it, int id)
+        {
+            Color selectedColor = colors[id % colors.Length];
+            Color adjustedColor = Color.FromArgb(selectedColor.R, selectedColor.G, selectedColor.B);
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: useless line

File: NNPTPZ1/Program.cs [hunk 6]
```diff
@@ -23,311 +10,159 @@ namespace NNPTPZ1
...
-            internal Cplx Divide(Cplx b)
-            {
-                // (aRe + aIm*i) / (bRe + bIm*i)
-                // ((aRe + aIm*i) * (bRe - bIm*i)) / ((bRe + bIm*i) * (bRe - bIm*i))
-                //  bRe*bRe - bIm*bIm*i*i
-                var tmp = this.Multiply(new Cplx() { Re = b.Re, Imaginari = -b.Imaginari });
-                var tmp2 = b.Re * b.Re + b.Imaginari * b.Imaginari;
+        private static void PaintPixel(Bitmap bitmap, int i, int j, Color selectedColor)
+        {
+            bitmap.SetPixel(j, i, selectedColor);
```
Comment (rodi0878) [inline NNPTPZ1/Program.cs]: you seriously messed up whole algorithm - now you use J for the witdth dimension, but before you used it with height in for loop. You need to recheck and fix every match between i,j - witdth,height dimensions

**Review by rodi0878 — CHANGES_REQUESTED** [review]
There are several unfinished problems and also very big bug...

**Comment by Axoden** [comment]
Changes made on the development branch according to your requests

**Comment by rodi0878** [comment]
ok, accepted

**Pull request closed without merging**